Course_Part_10/src/TicTacToeBasic.java

```diff
@@ -0,0 +1,247 @@
+    }
+
+    /**
+     * The board will be displayed according to the play mode chosen
+     * A: will show all the made moves and all the remaining options
+     * B: will only show all the made moves
+     */
+    static void displayBoard() {
+        String first;
+        String second;
+        String third;
+        System.out.println();
+        // Board will be printed layer by layer beginning from the top
+        for (int i = 1; i <= 3; i++) {
+            if (playMode.equals("A")) {
+                first = playOptions.get("A" + i).equals("  ") ? ("A" + i) : playOptions.get("A" + i);
+                second = playOptions.get("B" + i).equals("  ") ? ("B" + i) : playOptions.get("B" + i);
+                third = playOptions.get("C" + i).equals("  ") ? ("C" + i) : playOptions.get("C" + i);
+            } else {
+                first = playOptions.get("A" + i);
+                second = playOptions.get("B" + i);
+                third = playOptions.get("C" + i);
+            }
+
+            System.out.printf(" %s | %s | %s \n", first, second, third);
+            if (i != 3) {
+                System.out.println("--------------");
+            }
+        }
+        System.out.println();
+    }
+
+    /**
+     * Validate the player's input (move)
+     * A move is valid if it's one of the remaining options
+     *
+     * @param move is the player's input
+     * @return true if move is not valid and false is valid
+     */
+    static boolean isMoveNotValid(String move) {
+        ArrayList<String> remainingOptions = remainingPlayOptions();
+        if (!remainingOptions.contains(move)) {
+            System.out.println("This is an invalid move!");
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * The options (moves) that are still available to be played
+     *
+     * @return arraylist of the remaining options
+     */
+    static ArrayList<String> remainingPlayOptions() {
+        ArrayList<String> remainingOptions = new ArrayList<>();
+        playOptions.forEach((key, value) -> {
+                    if (value.equals("  ")) remainingOptions.add(key);
+                }
+        );
+        return remainingOptions;
+    }
+
+    /**
+     * The game consists out of 9 turns that can result in a win
+     * If turn 10 reached this means none of the players won
+     *
+     * @param turn count
+     * @return true if turn 10 is reached and it's a draw and false if not yet
+     */
+    static boolean isDraw(int turn) {
+        if (turn == 10) {
+            System.out.println("It’s a draw!");
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Check if player or computer won by having 3 of their marks in a line
+     * Win options: 3 horizontal lines - 3 vertical lines - 2 diagonal lines
+     *
+     * @return true if there is a winner and false if no winner yet
+     */
+    static boolean isWin() {
```

Comment:
Het bepalen van een winner kan efficiënter door niet 9x te loopen en daar binnen de switch op te nemen, maar door dat je ook binnen deze methode te print doet kan het niet anders.

Netter is om Separation of Concern toe te passen en de methode puur en alleen true of false terug te laten geven en in de aanroepende code (de main) het print werk te laten doen. Die zou ook weten wie aanzet is. 

Echter betekent het dat je je implementatie best wel veel moet aanpassen ;)